File: Java/Medium/WrongAnswer/find-original-array-from-doubled-array_7.java
class Solution {
    public int[] findOriginalArray(int[] changed) {
        
        if (changed.length % 2 != 0) {
            return new int[]{};
        }
        int[] result = new int[changed.length / 2];
        Arrays.sort(changed);
        int i = 0;
        int j = 0;
        while (i < changed.length) {
            if (i == changed.length - 1) {
                return new int[]{};
            }
            if (changed[i] != changed[i + 1]) {
                return new int[]{};
            }
            result[j++] = changed[i];
            i += 2;
        }
        return result;
    }
}

File: C++/Medium/WrongAnswer/find-original-array-from-doubled-array_7.cpp
class Solution {
public:
    vector<int> findOriginalArray(vector<int>& changed) {
        
        vector<int> result;
        if (changed.size() % 2 == 1) return result;
        map<int, int> count;
        for (auto c : changed) {
            count[c]++;
        }
        for (auto c : changed) {
            if (count[c] == 0) continue;
            if (count[c * 2] == 0) return vector<int>();
            count[c]--;
            count[c * 2]--;
            result.push_back(c);
        }
        return result;
    }
};